src/graphic/Fast3D/gfx_sdl2.cpp

```diff
@@ -223,20 +223,21 @@ static void set_fullscreen(bool on, bool call_callback) {
     if (fullscreen_state == on) {
         return;
     }
-    fullscreen_state = on;
     int display_in_use = SDL_GetWindowDisplayIndex(wnd);
-    SDL_DisplayMode mode;
     if (display_in_use < 0) {
-        SDL_GetDesktopDisplayMode(0, &mode);
-    } else {
-        SDL_GetDesktopDisplayMode(display_in_use, &mode);
+        SPDLOG_WARN("Can't detect on wich monitor we are. Probably out of display area?");
+        SPDLOG_WARN(SDL_GetError());
```

Comment:
Does this return a string or an error code? If it's an error code we might want to put some text here.

Reply:
SDL dogs says it is a string:
Returns a message with information about the specific error that occurred, [...]
https://wiki.libsdl.org/SDL2/SDL_GetError